apps/full-site-editing/full-site-editing-plugin/donations/edit.js

```diff
@@ -1,55 +1,63 @@
 /**
  * External dependencies
  */
-import classNames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { useState } from '@wordpress/element';
+import { useState, useEffect } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Button } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
+import DonationsTabs from './donations-tabs';
+import LoadingError from './loading-error';
+import LoadingStatus from './loading-status';
+import UpgradePlan from './upgrade-plan';
+import fetchStatus from './fetch-status';
 
 const Edit = () => {
-	const [ activeTab, setActiveTab ] = useState( 'one-time' );
-
-	const isActive = ( button ) => ( activeTab === button ? 'active' : null );
-
-	return (
-		<div className="donations__container">
-			<Button className={ isActive( 'one-time' ) } onClick={ () => setActiveTab( 'one-time' ) }>
-				{ __( 'One-Time', 'full-site-editing' ) }
-			</Button>
-			<Button className={ isActive( 'monthly' ) } onClick={ () => setActiveTab( 'monthly' ) }>
-				{ __( 'Monthly', 'full-site-editing' ) }
-			</Button>
-			<Button className={ isActive( 'annually' ) } onClick={ () => setActiveTab( 'annually' ) }>
-				{ __( 'Annually', 'full-site-editing' ) }
-			</Button>
-			<div
-				id="donations__tab-one-time"
-				className={ classNames( 'donations__tab', { active: isActive( 'one-time' ) } ) }
-			>
-				{ __( 'One time', 'full-site-editing' ) }
-			</div>
-			<div
-				id="donations__tab-monthly"
-				className={ classNames( 'donations__tab', { active: isActive( 'monthly' ) } ) }
-			>
-				{ __( 'Monthly', 'full-site-editing' ) }
-			</div>
-			<div
-				id="donations__tab-annually"
-				className={ classNames( 'donations__tab', { active: isActive( 'annually' ) } ) }
-			>
-				{ __( 'Annually', 'full-site-editing' ) }
-			</div>
-		</div>
-	);
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ loadingError, setLoadingError ] = useState( '' );
+	const [ shouldUpgrade, setShouldUpgrade ] = useState( false );
+	const [ upgradeUrl, setUpgradeUrl ] = useState( '' );
+
+	const mapStatusToState = ( result ) => {
+		if ( ( ! result && typeof result !== 'object' ) || result.errors ) {
+			setLoadingError( __( 'Could not load data from WordPress.com.', 'full-site-editing' ) );
+		} else {
+			setShouldUpgrade( result.should_upgrade_to_access_memberships );
+			setUpgradeUrl( result.upgrade_url );
+		}
+
+		setIsLoading( false );
+	};
+
+	const apiError = ( message ) => {
+		setLoadingError( message );
+		setIsLoading( false );
+	};
+
+	const updateData = () => {
+		fetchStatus().then( mapStatusToState, apiError );
+	};
+
+	useEffect( () => updateData() );
```

Comment:
Do we need to run this effect every time the component is updated or only on first mount? Seems [we do the latter in the Premium Content block](https://github.com/Automattic/wp-calypso/blob/ace20c7943553269eb836aec4217ca90b6a42835/apps/full-site-editing/full-site-editing-plugin/premium-content/blocks/container/edit.js#L280), so we'd likely want to do the same here:

```suggestion
	useEffect( () => updateData(), [] );
```

Reply:
> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

Ah clever, TIL! I missed that in docs: https://reactjs.org/docs/hooks-effect.html

Reply:
Oooh, nice: e0161e12c8 (`updateData` had to be moved inside the `useEffect` callback for optimization purposes).